### PR TITLE
Remove padding after pylate encoding

### PR DESCRIPTION
**Pull request by NohTow**
Hello,

As pointed out in the [original merging of PyLate ](https://github.com/embeddings-benchmark/mteb/pull/3183) and [the following discussion](https://github.com/embeddings-benchmark/mteb/issues/3283), after the encoding through PyLate models, there is an unnecessary padding operation because PyLate indexes handle padding on their own.

Originally, I observed some deltas, but lately when running new benches with new models, I actually have scores of 0, while with my scripts I had normal results. I checked the code and tested running with MTEB but without the padding and the scores are normal again. I cannot share a reproducible minimal code because the models are unfortunately not public for now but given that the padding is not needed and the code works great without it, I propose to remove it anyways.

I am not totally sure why this can totally _break_ some results, maybe it's the fact that we double cast it (here and in PyLate) or maybe I missed one effect of this particular padding call.

Close https://github.com/embeddings-benchmark/mteb/issues/3283

### CODE REVIEW

File: mteb/models/model_implementations/pylate_models.py [hunk 1]
```diff
@@ -328,13 +328,10 @@ def encode(
...
             prompt_name=prompt_name,
             is_query=prompt_type == PromptType.query,
-            convert_to_tensor=True,
             **kwargs,
         )
 
-        # encode returns a list of tensors shaped (x, token_dim), pad to uniform length
-        pred = torch.nn.utils.rnn.pad_sequence(pred, batch_first=True, padding_value=0)
-        return pred.cpu().numpy()
+        return pred
```
Comment (Samoed) [inline mteb/models/model_implementations/pylate_models.py:334]: Is it numpy be default, right?

Comment (NohTow) [reply mteb/models/model_implementations/pylate_models.py:334]: A list of numpy arrays, yes!

**Comment by Samoed** [comment]
From my test in https://github.com/embeddings-benchmark/mteb/issues/3283 padding didn't play any role. Only double casting caused difference

**Comment by NohTow** [comment]
Yeah probably, the second padding should be a no-op.
Both were not required in any case, thanks for merging!
